docker-images/build.sh

```diff
@@ -65,14 +66,61 @@ function build {
     do
         sha=${version_checksums[$kafka_version]}
         lib_directory=${version_libs[$kafka_version]}
+        
+        # If there is a file specified for this version of Kafka use that instead of the specified URL
+        if [ ${version_binary_files[$kafka_version]} ]
+        then 
+
+            filepath=${version_binary_files[$kafka_version]}
+            echo "Kafka binary file field was specified, using $filepath instead of URL"
+            
+            # Extract the filename from the filepath
+            OIFS=$IFS
+            IFS="/"
+            read -a fparr <<< $filepath
+            filename=${fparr[-1]}
+            IFS=$OIFS
+            
+        elif [ ${version_binary_urls[$kafka_version]} ]
+        then
+
+            url=${version_binary_urls[$kafka_version]}
+            echo "Downloading Kafka $kafka_version binaries from: $url"
+
+            # Extract the filename from the URL
+            OIFS=$IFS
+            IFS="/"
+            read -ra urlarr <<< "$url"
+            filename=${urlarr[-1]}
+            IFS=$OIFS
+            
+            filepath="$(mktemp -d)/$filename"
```

Comment:
Why use a temp directory? You could avoid needing to download between successive invocations by:

1. Downloading the tar.gz (each version) to it's own non-random file if that files doesn't already exist or doesn't match its checksum
2. Not deleting the file once you've used it in the image.

Reply:
Yeah that does make a lot more sense. My issue is with the loop through the `kafka_images` list. There only 2 folders in that list and the `kafka` image needs the binaries copying in and a lot more build args than the `test_client`. Would it be better to just separate the two and have a separate make command for each?

Reply:
What exactly is the problem? The annoyance of the unused build arg warning?

Reply:
No not the warnings, it was doing anything else in that loop which was specific to only the `kafka` image. But I think I have found a way around it now.